homeassistant/components/mobile_app/util.py

```diff
@@ -53,3 +57,20 @@ def get_notify_service(hass: HomeAssistant, webhook_id: str) -> str | None:
             return target_service
 
     return None
+
+
+_CLOUD_HOOK_LOCK = asyncio.Lock()
+
+
+@callback
+async def create_cloud_hook(
+    hass: HomeAssistant, webhook_id: str, entry: ConfigEntry | None
+) -> str:
+    """Create a cloud hook."""
+    async with _CLOUD_HOOK_LOCK:
```

Comment:
The only downside of using a util function is that we have only one lock and not a lock per config entry.
Is this fine or should we create a look per `webhook_id`?

Reply:
The deduplication required in the cloud client for webhook ids is Home Assistant instance wide since there's only one cloud client per instance. That makes me think it's good to have a single lock.